tests/ut/runtime/test_benchmark.py

```
@@ -770,6 +820,31 @@ def emit_dispatch(*_args: Any, **_kwargs: Any) -> None:
 
 
+def test_benchmark_l3_reports_setup_markers_without_dispatch(span_root):
+    """Captured setup markers are distinguished from no STRACE output."""
+    rt = _FakeDistributedWorker()
+
+    with (
+        patch.object(dcp_mod, "DistributedCompiledProgram", _FakeDistributedCompiled),
+        patch("pypto.runtime.bench.configure_log"),
+        patch("pypto.runtime.bench.current_level", return_value=20),
+        pytest.raises(
+            RuntimeError, match="captured 2 .* but none formed usable L3 dispatch timing"
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)